src/_lib/sass/src/_spacing.scss

```diff
@@ -0,0 +1,72 @@
+
```

Comment:
This look very scary and interesting at the same time! I have never used if else, for loops or @each in sass but its clear enough what its doing.